### PR TITLE
[CUDA] Add addmv for complex dtypes

### PR DESCRIPTION
**Pull request by anjali411**

### CODE REVIEW

File: aten/src/ATen/cuda/CUDABlas.h [hunk 1]
```diff
@@ -34,6 +34,10 @@ void gemm<double>(CUDABLAS_GEMM_ARGTYPES(double));
 template <>
 void gemm<float>(CUDABLAS_GEMM_ARGTYPES(float));
 template <>
+void gemv<c10::complex<double>>(CUDABLAS_GEMV_ARGTYPES(c10::complex<double>));
```
Comment (zasdfgbnm) [inline aten/src/ATen/cuda/CUDABlas.h:37]: I think you need to move this down to `gemv`, you are currently putting it to `gemm`.

**Comment by zasdfgbnm** [comment]
I remember I saw this PR before, was that one reverted due to windows failure?

**Comment by anjali411** [comment]
> I remember I saw this PR before, was that one reverted due to windows failure?

yeah

File: aten/src/ATen/cuda/CUDABlas.h (hunk 1 above)
Comment (anjali411) [reply aten/src/ATen/cuda/CUDABlas.h]: oh yeah my bad. updated

File: aten/src/ATen/native/cuda/Blas.cu [hunk 2]
```diff
@@ -12,7 +12,7 @@ Tensor &addmv_impl_cuda(Tensor& result, const Tensor &self, const Tensor &mat, c
   const auto vec_contiguous = vec_stride == 0 ? vec.contiguous() : vec;
   vec_stride = vec_contiguous.stride(0);
 
-  AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, mat.scalar_type(), "addmv_impl_cuda", [&] {
+  AT_DISPATCH_FLOATING_TYPES_AND_COMPLEX_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, mat.scalar_type(), "addmv_impl_cuda", [&] {
```
Comment (zasdfgbnm) [inline aten/src/ATen/native/cuda/Blas.cu:15]:
```suggestion
  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, mat.scalar_type(), "addmv_impl_cuda", [&] {
```

File: aten/src/ATen/cuda/CUDABlas.cpp [hunk 3]
```diff
@@ -298,6 +298,30 @@ void gemm<at::BFloat16>(CUDABLAS_GEMM_ARGTYPES(at::BFloat16)) {
     CUDABLAS_POSINT_CHECK(gemv<Dtype>, incy); \
   } while (0)
 
+template <>
+void gemv<c10::complex<double>>(CUDABLAS_GEMV_ARGTYPES(c10::complex<double>)) {
+  cublasHandle_t handle = at::cuda::getCurrentCUDABlasHandle();
```
Comment (zasdfgbnm) [inline aten/src/ATen/cuda/CUDABlas.cpp:303]: I believe you need to `ifdef` the function body for ROCm as well?

**Review by zasdfgbnm — APPROVED** [review]
LGTM, given that ROCm CI is not failing

**Pull request closed without merging**